### DISCUSSION TITLE
DLNA not working

### DISCUSSION THREAD

**Discussion by justaguyinaz** [stashapp/stash#5140] [Q&A] [answered]
I installed on my Synology via Docker and the app works great. I can enable DLNA and see the server on a client, but get errors when trying to access it. One error shows "Start tag expected, '>' not found". It seems to be working for actually broadcasting its availability, but beyond that, nothing. Any ideas?

**Comment by justaguyinaz** [comment]
If I open up a browser window to that port, this is what the screen shows. Big white page with the box on it.

![image](https://github.com/user-attachments/assets/6e84c049-223f-4b13-9633-59ff471382f9)

**Comment by WithoutPants** [comment]
What clients are you trying to access the DLNA with?

**Reply by justaguyinaz** [reply]
BigScreen and Skybox VR on the Oculus, nPlayer on my iPhone, etc.

**Comment by WithoutPants** [comment]
The DLNA service is finicky at the best of times unfortunately. I haven't tested on any of those clients. I've done a quick test using VLC and it appears to be working there. Make sure you've whitelisted the IP addresses your accessing from as well.

On what client are you getting the error in your original description? What is the behaviour on other clients?

**Reply by justaguyinaz** [reply, marked as answer]
Ahhh… the whitelisting is what I must have been missing. I saw the IPs showing up and thought that was sufficient. Didn’t know I needed to do that if the DLNA service itself was running. Once I added an IP, things worked fine. Thanks for the assist!

**Comment by vancrize** [comment]
**Stash in Unraid Docker container in host mode**

I have a similar situation, but dont even see client IPs that i can whitelist.

- No other app is interfering with the port
- Other DLNA Providers like Plex are found (I use DeoVR on different platforms)
- In Win 11 network window the stash DLNA shows up 
![image](https://github.com/user-attachments/assets/4d56545b-f235-4845-909d-397e9b6f954d)

- If i open the :1900 in browser i get only the little form, too 
![image](https://github.com/user-attachments/assets/0f1f5c4e-5a45-4dd4-9111-a5997abaf0e3)

- I added client IPs to the whitelist manually, as well as trying the wildcard, no difference
- From this thread here i assume clients that try to connect should show up at the top of /settings?tab=services to whitelist from there, but no client ever shows up there, even the win11 one, that sees something in network window or browser.

I tried different container networks modes and lots of port forwarding, but the result is always the same: something DLNA related is there, but no browsable endpoint.

Any ideas?

**Comment by feederbox826** [comment]
> Other DLNA Providers

only one DLNA provider per platform, it is exclusive to the machine, not ports

**Reply by vancrize** [reply]
Even if it's the only running docking container on the whole NAS, the result is the exact same, in every network mode, with every port i could imagine and even with its own IP. The fact that it is broadcasted in this weird broken way, tells me that only a little step seems to be missing.

**Reply by feederbox826** [reply]
Does it work if you disable plex on the machine?

**Reply by vancrize** [reply]
As I said. Even if it's the only running container on the whole system, the outcome is exactly the same as when Plex and even jellyfin broadcast their DLNA. I rebooted the whole NAS with not autostarting any docker but stash, its the same result: just the empty form in the browser and no DLNA in video clients.

**Reply by feederbox826** [reply]
> even the win11 one, that sees something in network window or browser.

You have to connect into the DLNA server and get rejected in order to add it to the whitelist, showing up and scanning does not get you on the list

**Reply by vancrize** [reply]
How do i "connect" into the server to trigger the whitelist? My usual clients like VLC on different platforms, VR Headset, DeoVR and Smart TVs don't even see the server, but will see every other DLNA broadcast like my 2 Plex and one I broadcasted just for testing from a CFW PS3.

It's only due to the network window in windows and the browser access i was able to confirm that at least something is running when I activate DLNA.
